### PR TITLE
Ensure parent directories are readable by target user when creating directories in `config apply`

### PR DESCRIPTION
**Pull request by arsing**
When `config apply` creates the /var/secrets/aziot/keyd directory to store
the inline symmetric key, it assumed that parent directories would already
be readable by the target user. But this is not a valid assumption since
it depends on root's umask, and we've had a few cases of devices where
root's umask is such that new directories are 0700 or 0750 by default.

This change adds a step of walking through the parents,
checking if they're readable by the target user, and making them
world-readable if they're not.

### CODE REVIEW

**Comment by gordonwang0** [comment]
Can we also add a check for this in the read_key_pairs check, since it's possible to modify the directories' permissions after creation?

**Review by gordonwang0 — APPROVED** [review]

**Comment by arsing** [comment]
>Can we also add a check for this in the read_key_pairs check, since it's possible to modify the directories' permissions after creation?

Done, and for certs checks too.

File: aziotctl/src/internal/check/checks/read_key_pairs.rs [hunk 1]
```diff
@@ -62,61 +62,79 @@ impl ReadKeyPairs {
...
+        let aziotks_user = crate::internal::common::get_system_user("aziotks")?;
+
+        for (id, path) in preloaded_keys {
+            let mut readable = true;
+            if let Ok(aziot_keys_common::PreloadedKeyLocation::Filesystem { path }) = path.parse() {
+                if let Err(err) =
+                    aziotctl_common::config::check_readable(&path, &aziotks_user, false)
+                {
+                    err_aggregated.push(format!("{:?}", err));
+                    readable = false;
```
Comment (gordonwang0) [inline aziotctl/src/internal/check/checks/read_key_pairs.rs:75]: I don't think it's necessary to keep a `readable` flag? You can just `continue` when not readable.

**Review by gordonwang0 — APPROVED** [review]

File: aziotctl/src/internal/check/checks/read_key_pairs.rs (hunk 1 above)
Comment (arsing) [reply aziotctl/src/internal/check/checks/read_key_pairs.rs:75]: Ah yes. It was necessary when I wrote it, but then I refactored it a little and now it's not necessary. Removed.

**Pull request merged**